llvm/lib/ObjectYAML/GOFFYAML.cpp

```diff
@@ -12,34 +12,92 @@
 
 #include "llvm/ObjectYAML/GOFFYAML.h"
 #include "llvm/BinaryFormat/GOFF.h"
-#include <string.h>
 
 namespace llvm {
-namespace GOFFYAML {
 
-Object::Object() {}
+namespace yaml {
 
-} // namespace GOFFYAML
+void ScalarEnumerationTraits<GOFFYAML::GOFF_AMODE>::enumeration(
+    IO &IO, GOFFYAML::GOFF_AMODE &Value) {
+#define ECase(X) IO.enumCase(Value, #X, GOFF::ESD_##X)
+  ECase(AMODE_None);
+  ECase(AMODE_24);
+  ECase(AMODE_31);
+  ECase(AMODE_ANY);
```

Comment:
Can it be changed to `AMODE_Any`? It is for consistency with `AMODE_None`.

Reply:
No. AMODE_None means unspecified, and that means it defaults to AMODE 24. AMODE_ANY means either AMODE 24 or AMODE 31. These are different bit patterns in the object and thus cannot be collapsed without throwing away data from the object.

Reply:
OK, I see. Thanks.